clang/test/utils/update-verify-tests/Inputs/duplicate-diag.c

```diff
@@ -0,0 +1,10 @@
+// RUN: %clang_cc1 -verify %s
+// RUN: diff %s %s.expected
```

Comment:
The way I understand it, the update-verify-tests plugin in lit will overwrite this source file, which may end up overwriting this RUN line if there is a bug. Basically, there is a weird circularity here.

Therefore, it seems to me that the RUN lines for the `diff` against expected ought to be in the `lit-plugin.test` file instead.

Reply:
The files in the `Inputs` directory are not discovered by lit, so this doesn't actually run the RUN lines. Instead a copy is made, and a new lit instance is launched to test the run lines.

This is in `duplicate-diag.test`, the RUN lines of `duplicate-diag.c` are not executed at all for this case because it tests the `update-verify-tests.py` script:
```
# RUN: cp %S/Inputs/duplicate-diag.c %t.c && not %clang_cc1 -verify %t.c 2>&1 | %update-verify-tests
# RUN: diff -u %S/Inputs/duplicate-diag.c.expected %t.c
# RUN: %clang_cc1 -verify %t.c
```

This is in `lit-plugin.test`, which copies the files to a separate directory and launches llvm-lit there to test the lit integration:
```
# RUN: cp %S/Inputs/*.c %S/LitTests
# RUN: cp %S/Inputs/*.c.expected %S/LitTests
# RUN: not %{lit} %S/LitTests
# RUN: not %{lit} %S/LitTests --update-tests
# RUN: %{lit} %S/LitTests
```

Reply:
Right, it's all a little mind-bending, but my point is that since you're trying to test whether the invocation of `not %{lit} %S/LitTests --update-tests` in `lit-plugin.test` updated the tests to look like the `.expected` file, that invocation of `diff` should *also* be in `lit-plugin.test`.

You can't have the correctness check for a test that modifies a file be in the same file that is being modified. It has to live outside.